ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt

```diff
@@ -15,65 +15,53 @@ import io.ktor.util.cio.*
 import io.ktor.util.pipeline.*
 import io.ktor.utils.io.*
 import kotlinx.coroutines.*
-import kotlin.coroutines.*
 
 /**
  * The default minimal content size to compress
  */
 internal const val DEFAULT_MINIMAL_COMPRESSION_SIZE: Long = 200L
 
-/**
- * Compression plugin configuration
- */
-public data class CompressionOptions(
-    /**
-     * Map of encoder configurations
-     */
-    val encoders: Map<String, CompressionEncoderConfig> = emptyMap(),
-    /**
-     * Conditions for all encoders
-     */
-    val conditions: List<ApplicationCall.(OutgoingContent) -> Boolean> = emptyList()
-)
+private object ContentEncoding : Hook<suspend ContentEncoding.Context.(ApplicationCall) -> Unit> {
 
-/**
- * Configuration for an encoder
- */
-public data class CompressionEncoderConfig(
-    /**
-     * Name of the encoder, matched against entry in `Accept-Encoding` header
-     */
-    val name: String,
-    /**
-     * Encoder implementation
-     */
-    val encoder: CompressionEncoder,
-    /**
-     * Conditions for the encoder
-     */
-    val conditions: List<ApplicationCall.(OutgoingContent) -> Boolean>,
-    /**
-     * Priority of the encoder
-     */
-    val priority: Double
-)
+    class Context(private val pipelineContext: PipelineContext<Any, ApplicationCall>) {
+        fun transformBody(block: (OutgoingContent) -> OutgoingContent?) {
```

Comment:
What happens if ` OutgoingContent?` is `null`?

Reply:
It means that the body could not be transformed and the plugin will keep the old body